### DISCUSSION TITLE
Get full page content including iFrames (for node version)

### DISCUSSION THREAD

**Discussion by adnan-ahmed** [GoogleChrome/lighthouse#13108] [Q&A]
Hi!
I understand that we can get page content from the artifacts.json after running lighthouse (artifacts.json[MainDocumentContent]). However, I was wondering if there's a way to dump the HTML content from the iframes as well in the website. If not, anyone please point me to how I could implement it?

best

**Comment by adamraine** [comment]
I don't think this is currently possible. We do have an artifact for iframe elements, but it doesn't contain the iframe contents. You could look at modifying the code yourself though:

https://github.com/GoogleChrome/lighthouse/blob/b83573be5ecb20d996238c961e9a3e70e2f6c72b/lighthouse-core/gather/gatherers/iframe-elements.js#L19-L35
